src/RateLimit/RateLimitManager.php

```diff
@@ -41,29 +55,32 @@ public function getAccount() {
   }
 
   /**
-   * Set the plugin info.
+   * Constructor for RateLimitManager.
    *
-   * @param array $pluginInfo
-   */
-  public function setPluginInfo($pluginInfo) {
-    $this->pluginInfo = $pluginInfo;
-  }
-
-  /**
-   * Constructor for RestfulRateLimitManager.
-   *
-   * @param string $resource
-   *   Resource name being checked.
-   * @param array $plugin
-   *   The plugin info array for the rate limit.
-   * @param \stdClass $account
+   * @param \RestfulBase $resource
+   *   Resource being checked.
+   * @param array $plugin_options
+   *   Array of options keyed by plugin id.
+   * @param object $account
    *   The identified user account for the request.
+   * @param RateLimitPluginManager $manager
+   *   The plugin manager.
    */
-  public function __construct($resource, array $plugin, $account = NULL) {
-    parent::__construct($plugin);
+  public function __construct(\RestfulBase $resource, $plugin_options, $account = NULL, RateLimitPluginManager $manager = NULL) {
     $this->resource = $resource;
-    $this->setPluginInfo($plugin);
     $this->account = $account ? $account : drupal_anonymous_user();
+    $manager = $manager ?: RateLimitPluginManager::create();
+    $options = array();
+    foreach ($plugin_options as $plugin_id => $rate_options) {
+      // Set the instance id to articles::request and specify the plugin id.
+      $instance_id = $resource->getResourceName() . '::' . $plugin_id;
+      $options[$instance_id] = array(
+        'id' => $plugin_id,
+        'resource' => $resource,
+      );
+      $options[$instance_id] += $rate_options;
+    }
+    $this->plugins = new RateLimitPluginCollection($manager, $options);
```

Comment:
From the code documents:
This is the D8 recommended way of dealing with arrays of plugins.

> A plugin collection is used to contain plugins that will be lazily instantiated. The configurations of each potential plugin are passed in, and the configuration key containing the plugin ID is specified by self::$pluginKey.